dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -1054,37 +1054,37 @@ void InterpreterSelectQuery::executeImpl(TPipeline & pipeline, const BlockInputS
             options.to_stage > QueryProcessingStage::WithMergeableState &&
             !query.group_by_with_totals && !query.group_by_with_rollup && !query.group_by_with_cube;
 
-        if (expressions.first_stage)
+        if (expressions.filter_info)
```

Comment:
Would it work if data comes in intermediate processing stage from Distributed table?

Reply:
I guess it won't - for aggregations we need to setup row policies on remote servers too. We don't have options to propagate row policy with DDL or by any other means. Am I correct?

Reply:
It would work, if the remote shards also have properly setup row policies. Added test.